main.go

```diff
@@ -68,7 +68,9 @@ func CreateSnapshotRepository(elasticURL, repoType, bucketName, username, passwo
 	client := &http.Client{Transport: tr}
 	url := fmt.Sprintf("%s://%s:9200/_snapshot/%s", scheme, elasticURL, bucketName)
 	body := fmt.Sprintf("")
-	if repoAccessKey!="" && repoSecretKey!="" {
+	if repoType=="azure" {
+		body = fmt.Sprintf("{ \"type\": \"%s\", \"settings\": { \"container\": \"%s\", \"compress\": \"true\"  } }", repoType, bucketName)
+	} else if repoAccessKey!="" && repoSecretKey!="" {
```

Comment:
We should standardize on the check. If we have a flag to determine repo-type, then let's make that consistent.

Reply:
Just pushed a commit, let me know if I understood what you were asking for, if not could you provide a little more direction.  Thanks!